src/app/components/HeaderAuth.tsx

```diff
@@ -0,0 +1,41 @@
+'use client';
+
+import { useSession } from 'next-auth/react';
+import * as actions from '@/actions';
+
+export default function HeaderAuth() {
+  const session = useSession();
+
+  let authContent: React.ReactNode;

```

Comment:
You can refactor this whole block to remove `authContent` and just return null or the elements when the condition is met